finetune/target_models/document_labeling.py

```diff
@@ -0,0 +1,107 @@
+from finetune.target_models.sequence_labeling import SequenceLabeler, SequencePipeline
+from finetune.encoding.input_encoder import EncodedOutput
+
+def get_context(document, dpi_norm):
+    """
+    Gets the context as formatted for the DocRep base model from 
+    the output of indico's PDFExtraction api.
+    """
+    context = []
+    for page in document:
+        # This norm factor is 
+        if dpi_norm:
+            dpi = page["pages"][0]["dpi"]
+            x_norm = 300 / dpi["dpix"]
+            y_norm = 300 / dpi["dpiy"]
+        else:
+            x_norm = 1.
+            y_norm = 1.
+
+        for token in page["tokens"]:
+            pos = token["position"]
+            offset = token["doc_offset"]
+            context.append(
+                {
+                    'top': pos["top"] * y_norm,
+                    'bottom': pos["bottom"] * y_norm,
+                    'left': pos["left"] * x_norm,
+                    'right': pos["right"] * x_norm,
+                    'start': offset["start"],
+                    'end': offset["end"],
+                    'text': token["text"],
+                }
+	    )
+    return context
+
+def _single_convert_to_finetune(*, document, dpi_norm=True):
+    context = get_context(document, dpi_norm)
+    texts = []
+    offsets = []
+    last_end = -1
+    for page in document:
+        page_obj = page["pages"][0]
```

Comment:
ah this is where you were talking about with the page splitting, right? awesome thanks for taking this into consideration :)

Reply:
https://github.com/IndicoDataSolutions/finetune/pull/574/files/0dad7a31722a272a090a6db04c3c2ea22335ee57#diff-9ff8c4ad7622224166854f0e2af0bed4R88 yep, and this is the block that then generates the chunks from within the pages in a way that the de-chunking code used at predict can reconstruct.